### PR TITLE
Alerting: Declare incident from a firing alert

### PR DESCRIPTION
**Pull request by gillesdemey**
Continuation of https://github.com/grafana/grafana/pull/58419 where implementation has been moved back to the alerting code-base.

# What

This PR adds 3 things; a new `PluginBridge` component, `usePluginBridge` hook and a button in Alerting that uses it to render a `Declare Incident` button if the alert rule is firing.

It allows conditional rendering of content based on the existence of a plugin.

It also contains a very simple `createBridgeURL` function to help construct plugin URLs.

# How

```jsx
import { PluginBridge, BridgeSupportedPlugin } from '@grafana/runtime';

<PluginBridge 
  plugin={BridgeSupportedPlugin.Incident} 
  loadingComponent={<>Loading...</>}
  notInstalledComponent={<>Plugin not installed :(</>}
>
  The plugin is installed!
</PluginBridge>
```

or as a hook

```jsx
const { loading, installed, settings } = usePluginBridge(SupportedPlugin.Incident);

return (
  <>
    {loading === true && (
      <>Loading...</>
    )}
    {installed === false && (
      <>X is not installed</>
    )}
    {settings && (
      <>It seems to be installed!</>
    )}
  </>
);
```

I'm happy to bikeshed on the interface and props, so far the only required prop is `plugin` which needs a plugin identifier (from `plugin.json`).

image

### CODE REVIEW

File: public/app/features/plugins/pluginSettings.ts [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+import { BackendSrvRequest, getBackendSrv } from '@grafana/runtime';
 
 type PluginCache = {
   [key: string]: PluginMeta;
 };
 
 const pluginInfoCache: PluginCache = {};
 
-export function getPluginSettings(pluginId: string): Promise<PluginMeta> {
+export function getPluginSettings(pluginId: string, options?: Partial<BackendSrvRequest>): Promise<PluginMeta> {
```
Comment (gillesdemey) [inline public/app/features/plugins/pluginSettings.ts:10]: We're passing options here so we can disable the `showErrorAlert` property – if we don't then the user will see a error popup if a particular plugin is not installed.

File: packages/grafana-runtime/package.json [hunk 2]
```diff
@@ -57,15 +56,13 @@
     "@testing-library/user-event": "14.4.3",
     "@types/angular": "1.8.4",
     "@types/history": "4.7.11",
-    "@types/jest": "^29.2.2",
+    "@types/jest": "28.1.6",
```
Comment (VikaCep) [inline packages/grafana-runtime/package.json]: Why do you need to downgrade `jest`?

**Review by VikaCep — APPROVED** [review]
LGTM!

File: packages/grafana-runtime/package.json (hunk 2 above)
Comment (gillesdemey) [reply packages/grafana-runtime/package.json]: argh, this one was from the older PR and after merging with main it probably overwrote this 😓 – will revert

**Review by konrad147 — APPROVED** [review]
LGTM!, a few non-blocking nits

File: public/app/features/alerting/unified/components/PluginBridge.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+interface PluginBridgeHookResponse {
+  loading: boolean;
+  installed?: boolean;
+  error?: Error;
+  settings?: PluginMeta<{}>;
+}
+
+export interface PluginBridgeProps {
```
[the remark this answers is not in the document]
Comment (gillesdemey) [reply public/app/features/alerting/unified/components/PluginBridge.tsx:32]: Oops, totally right – must have occurred during a rebase with main :D

**Pull request merged**